src/map/include/items/inventory.h

```diff
@@ -18,10 +19,13 @@ enum class ReturnValue {
 
 // returns the first available spot in the inventory or 0 if there is no more space
 size_t get_first_available_spot(const EntitySystem& entitySystem, RoseCommon::Entity entity, RoseCommon::Entity item = entt::null);
+uint8_t has_bullets(const EntitySystem& entitySystem, RoseCommon::Entity entity);
+bool is_bullet_weapon(const EntitySystem& entitySystem, RoseCommon::Entity entity);
 ReturnValue add_item(EntitySystem&, RoseCommon::Entity entity, RoseCommon::Entity item);
 // will be entt::null if there is no item at that position
 RoseCommon::Entity remove_item(EntitySystem&, RoseCommon::Entity entity, size_t pos, uint32_t quantity);
```

Comment:
If we want to remove an item from an entity without dealing with slots, we will need to change this function to take an entity instead of a position. I did a quick search and it looks like we don't use this method outside of the inventory, so we could move this one inside `inventory.cpp` (without public exposure) and then add a new method `remove_item(EntitySystem&, Entity entity, Entity item, quantity)` which would find the slot and then call the internal function `remove_item` that uses the slot.